### ISSUE TITLE
Support for range addresses in 'First Sheet'!A1:'First Sheet'!A2 form

### ISSUE THREAD

**Issue by Jonathan--V** [ClosedXML/ClosedXML#1599]
## Read and complete the full issue template

Do not randomly delete sections. They are here for a reason.

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature
- [ ] Question

**Did you test against the latest CI build?**
- [ ] Yes
- [x] No

If you answered `No`, please test with [the latest development build](https://ci.appveyor.com/project/ClosedXML/ClosedXML/branch/develop/artifacts) first.

Where could I find step-by-step instructions on how to do this?

**Version of ClosedXML**

0.95.4.0

**What is the current behavior?**

Suppose a cell A1 uses a sheet name for specifying both the start and the end of the range in its formula. A NameNotRecognizedException is thrown when the value of A1 is calculated. Specifically, it is thrown on line 479 of CalcEngine.cs

**What is the expected behavior or new feature?**

No exception is thrown and the value of the cell is calculated correctly.

**Is this a regression from the previous version?**

Not as such. An uncaught exception is also thrown in 0.94.2.0, but this time it is an ExpressionParseException that happens earlier, on the access of cell.Value. Specifically, it is thrown on line 776 of CalcEngine.cs.

## Reproducibility
**This is an important section. Read it carefully. Failure to do so will cause a 'RTFM' comment.**

Without a code sample, it is unlikely that your issue will get attention. Don't be lazy. Do the effort and assist the developers to reproduce your problem. Code samples should be [minimal complete and verifiable](https://stackoverflow.com/help/mcve). Sample spreadsheets should be attached whenever applicable. Remove sensitive information.

**Code to reproduce problem:**
```c#
using System.Diagnostics;
using ClosedXML.Excel;

namespace ClosedXmlIssues
{
    public static class RangeWithBothEndsFromSheetName
    {
        public static void Main()
        {
            string filename = @"C:\ClosedXML and ExcelNumberFormat\Handling range from sheet name.xlsx";
            XLWorkbook workbook = new XLWorkbook(filename);
            IXLCell cell = workbook.Worksheet(2).Cell(1, 1);
            Debug.WriteLine($"{cell.Address}: {cell.Value}");
            workbook.CalculationOnSave = true;
            workbook.Save();
        }
    }
}

```

- [x] I attached a sample spreadsheet.  (You can drag files on to this issue)

[Handling range from sheet name.xlsx](https://github.com/ClosedXML/ClosedXML/files/5875006/Handling.range.from.sheet.name.xlsx)

**Comment by igitur** [comment]
Again, really non-standard way to specify the formula. 

You have
`=SUM('First Sheet'!A1:'First Sheet'!A2)`
instead of the more standard way:
`=SUM('First Sheet'!A1:A2)`

I can maybe fix this one easily though. But you're going to keep hitting problems in ClosedXML using non-standard conventions. We just don't have the manpower that the team behind Excel has.

**Comment by Jonathan--V** [comment]
I would also have written it only specifying the sheet name once, but I haven't created the workbooks that need to be read in. The authors have opted to use the longer formula, for whatever reason.

I understand your time is very limited - it would be nice to have this fixed if/when it fits in.

**Comment by jahav** [comment]
The range operator (the colon between two references) has been implemented in PR #1848 (part of #1812).

The range operator (like the reported `'First Sheet'!A1:'First Sheet'!A2`) now evaluate both arguments and creates a a new range from them (or error, if from different worksheets).
